### DISCUSSION TITLE
Signing through the browser using Yubikey

### DISCUSSION THREAD

**Discussion by Matt561** [openpgpjs/openpgpjs#1223] [Q&A] [answered]
I'm trying to connect a hardware device (Yubikey 5ci) to the browser to perform signing using the device's stored private key.

 I've attempted to use the WebUsb API but it seems that Yubico is on a chrome block list which means I am unable to target it (even though I'm using the correct vendorID/deviceID).

I've also tried to use WebAuthn to sign but it seems there is no way to choose which key I would like to use on the device. I can't for the life of me get it to sign using the secp256k1 curve. 

Wondering if this library would allow support for what I am doing.

Thanks any advance for any helpful ideas.

**Comment by dunxen** [comment, marked as answer]
Seems after [this](https://www.yubico.com/blog/webusb-and-responsible-disclosure/) WebUSB vulnerability, they might have blocked it, so I'm guessing you can only use it for FIDO U2F right now. I'd be interested too if this were not the case. Maybe there will be better support in the future again.

I don't think this library would help 😕

I know Ledger and Trezor use WebUSB to allow signing transactions. Guessing you could do that with arbitrary data.
